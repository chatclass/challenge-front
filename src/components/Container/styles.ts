import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: ${theme.breakpoints.md};
    margin: 0 auto;

    @media (min-width: ${({ theme }) => theme.breakpoints.xs}) {
      width: 100%;
    }
  `}
`
