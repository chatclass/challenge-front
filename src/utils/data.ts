/* eslint-disable @typescript-eslint/no-explicit-any */
const LIMIT = 6

export const normalizeData = (results: any) => {
  return {
    city: results.city,
    temperature: `${results.temp}°C ${results.description}`,
    min: `${
      results.forecast.find((f: any) => results.date.includes(f.date)).min
    }°`,
    max: `${
      results.forecast.find((f: any) => results.date.includes(f.date)).max
    }°`,
    wind: results.wind_speedy,
    humidity: `${results.humidity}%`,
    forecast: results.forecast
      .map((cast: any) => ({
        day: cast.weekday,
        min: `${cast.min}°`,
        max: `${cast.max}°`
      }))
      .filter((_cast: any, index: number) => index <= LIMIT)
  }
}
