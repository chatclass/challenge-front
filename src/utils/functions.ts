import { forecastDay } from '../store/ducks/weather/types';

export const unique = (
  value: forecastDay,
  index: number,
  self: forecastDay[],
) => {
  return (
    self
      .map(e => new Date(e.dt_txt).getDay())
      .indexOf(new Date(value.dt_txt).getDay()) === index
  );
};

export const capitalize = (str: string) => {
  const lower = str.toLowerCase();
  return str.charAt(0).toUpperCase() + lower.slice(1);
};
