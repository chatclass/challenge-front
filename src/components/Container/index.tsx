import React, { ReactNode } from 'react'
import * as S from './styles'

type ContainerProps = {
  children: ReactNode | ReactNode[]
}
export default function Container({ children }: ContainerProps) {
  return <S.Wrapper>{children}</S.Wrapper>
}
