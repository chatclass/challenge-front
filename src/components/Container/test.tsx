import React from 'react'
import { render, screen } from '../../utils/test-utils'

import Container from '.'

describe('<Container />', () => {
  it('should render the Container with a children', () => {
    const { container } = render(<Container>Children</Container>)

    screen.getByText('Children')

    expect(container.firstChild).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the Container with an element', () => {
    const { container } = render(
      <Container>
        <dl>
          <dt>Head</dt>
          <dd>Body</dd>
        </dl>
      </Container>
    )

    screen.getByText('Head')
    screen.getAllByText('Body')

    expect(container.firstChild).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
