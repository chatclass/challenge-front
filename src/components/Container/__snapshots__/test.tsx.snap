// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render the Container with a children 1`] = `
.c0 {
  width: 768px;
  margin: 0 auto;
}

@media (min-width:320px) {
  .c0 {
    width: 100%;
  }
}

<div
  class="c0"
>
  Children
</div>
`;

exports[`<Container /> should render the Container with an element 1`] = `
.c0 {
  width: 768px;
  margin: 0 auto;
}

@media (min-width:320px) {
  .c0 {
    width: 100%;
  }
}

<div
  class="c0"
>
  <dl>
    <dt>
      Head
    </dt>
    <dd>
      Body
    </dd>
  </dl>
</div>
`;
